cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -349,64 +355,63 @@ private void generateHtmlReport(Project project, PrintStream out, TestReport tes
         }
     }
 
-    private int runTestSuit(Path moduleTestCache, Target target, Collection<Path> testDependencies,
-                            Module module) throws IOException, InterruptedException {
+
+    private int runTestSuit(List<String> moduleNames, Path testCachePath, Target target, String packageName,
+                            String orgName) throws IOException,
+            InterruptedException {
         List<String> cmdArgs = new ArrayList<>();
         cmdArgs.add(System.getProperty("java.command"));
         String mainClassName = TesterinaConstants.TESTERINA_LAUNCHER_CLASS_NAME;
-        String orgName = module.packageInstance().packageOrg().toString();
-        String packageName = module.packageInstance().packageName().toString();
-        String moduleName = module.isDefaultModule() ? "" : module.moduleName().moduleNamePart();
-
-        String jacocoAgentJarPath = Paths.get(System.getProperty(BALLERINA_HOME)).resolve(BALLERINA_HOME_BRE)
-                .resolve(BALLERINA_HOME_LIB).resolve(TesterinaConstants.AGENT_FILE_NAME).toString();
-
 
         if (coverage) {
+            String jacocoAgentJarPath = Paths.get(System.getProperty(BALLERINA_HOME)).resolve(BALLERINA_HOME_BRE)
+                    .resolve(BALLERINA_HOME_LIB).resolve(TesterinaConstants.AGENT_FILE_NAME).toString();
             String agentCommand = "-javaagent:"
                     + jacocoAgentJarPath
                     + "=destfile="
                     + target.getTestsCachePath().resolve(TesterinaConstants.COVERAGE_DIR)
                     .resolve(TesterinaConstants.EXEC_FILE_NAME).toString();
+
             if (!TesterinaConstants.DOT.equals(packageName) && this.includesInCoverage == null) {
                 // add user defined classes for generating the jacoco exec file
                 agentCommand += ",includes=" + orgName + ".*";
             } else {
                 agentCommand += ",includes=" + this.includesInCoverage;
             }
+
             cmdArgs.add(agentCommand);
         }
 
-        String classPath = getClassPath(testDependencies);
-        cmdArgs.addAll(Lists.of("-cp", classPath));
+        cmdArgs.addAll(Lists.of("-cp", System.getProperty("java.class.path")));
+
         if (isInDebugMode()) {
             cmdArgs.add(getDebugArgs(this.err));
         }
         cmdArgs.add(mainClassName);
-        cmdArgs.add(moduleTestCache.toString());
+
+        String moduleNameList = "";
+        for (String moduleName : moduleNames) {
+            moduleNameList += moduleName + "#";
```

Comment:
Why do we have to append a `#` character here?

Reply:
Removed this implementation. Instead read the modules from the `test_suite.json` file at runtime